common/types/activation.go

```diff
@@ -169,46 +153,41 @@ func (m *ATXMetadata) MarshalLogObject(encoder log.ObjectEncoder) error {
 // ActivationTx is a full, signed activation transaction. It includes (or references) everything a miner needs to prove
 // they are eligible to actively participate in the Spacemesh protocol in the next epoch.
 type ActivationTx struct {
-	InnerActivationTx
+	NIPostChallenge
+	Coinbase Address
+	NumUnits uint32
+
+	NIPost   NIPost
```

Comment:
Does the Domain type for `ActivationTx` need to contain the full `NIPostChallenge`? I think just `PrevATXID` should be enough - all the other values are only needed to verify the `NIPost` which is only done by the `activation.Handler` and also doesn't need to be included into the domain type I believe.

Reply:
`Handler` in my opinion shouldn't use the domain type, but the wire type. It validates that the received ATX is valid before converting it to the domain type end then storing it in the DB. This would require fewer code changes and would allow to remove fields in `types.ActivationTx` immediately (if we want to keep that type around instead of creating a new domain type)

Reply:
> `Handler` in my opinion shouldn't use the domain type, but the wire type. It validates that the received ATX is valid before converting it to the domain type end then storing it in the DB. 

That's right! :) 

> This would require fewer code changes 

I'm not sure about this. Removing fields like NipostChallenge (i.e. the POST proof) would require lots of changes in different places (another code like looking up a positioning ATX depends on it) and I think it would end up with a bigger PR. 

Please, let's not waste time arguing about the most optimal way of introducing the changes - I'm sure there are plenty of ways and we can spend a day talking about them. I think we more-less agree about the desired outcome we want to achieve.